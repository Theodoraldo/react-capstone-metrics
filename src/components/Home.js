import React, { useState } from 'react';
import './style/Home.css';
import { Link } from 'react-router-dom';
import { BsArrowRightCircle } from 'react-icons/bs';
import europe from '../assets/europe.svg';
import afirca from '../assets/africa.svg';
import asia from '../assets/asia.svg';
import america from '../assets/america.svg';
import oceania from '../assets/oceania.svg';
import antarctic from '../assets/antarctica.svg';

const continentList = [
  { id: 1, continent: 'Africa', map: afirca },
  { id: 2, continent: 'America', map: america },
  { id: 3, continent: 'Antarctic', map: antarctic },
  { id: 4, continent: 'Asia', map: asia },
  { id: 5, continent: 'Europe', map: europe },
  { id: 6, continent: 'Oceania', map: oceania },
];

export default function Home() {
  const [search, setSearch] = useState('');
  return (
    <>
      <div className="home-header-container">
        <p className="home-header">The World&#8217;s Continent</p>
      </div>
      <div className="search-head">
        <input type="text" className="search" placeholder="Search country here ...." onChange={(e) => setSearch(e.target.value)} />
      </div>
      <div className="wrap">
        {
          continentList.filter((continent) => (
            search.toLowerCase() === '' ? continent : continent.continent.toLowerCase().includes(search)
          )).map((continent) => (
            <div key={continent.id} className="continent">
              <div className="cont-head">
                <div>{continent.continent}</div>
                <button type="button" className="arrow-right">
                  <Link to={`continent/${continent.continent}`}><BsArrowRightCircle size="1.5rem" color="#ececec" /></Link>
                </button>
              </div>
              <div className="cont-body">
                <p className="cont-name">{continent.continent}</p>
                <img src={continent.map} alt="continent logo" width={200} />
              </div>
            </div>
          ))
        }
      </div>
    </>
  );
}
