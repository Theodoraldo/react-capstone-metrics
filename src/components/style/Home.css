.continent {
  background-color: rgba(196, 124, 124, 0.2);
  border: 1px solid rgb(32, 31, 31);
  height: 200px;
  width: 360px;
  margin: 20px auto;
  padding: 10px;
  border-radius: 5px;
  box-shadow: 0 10px 35px -20px white;
}

.home-header-container {
  background-color: rgb(29, 27, 27);
  padding: 10px;
}

.home-header {
  text-align: center;
  font-size: 20px;
  font-weight: bold;
}

.search-head {
  position: sticky;
  top: 0;
}

.cont-head {
  font-size: 10px;
  display: flex;
  justify-content: space-between;
  margin-top: 10px;
}

.cont-name {
  font-size: 24px;
  font-weight: bold;
}

.cont-body {
  display: flex;
  gap: 10px;
}

.link-detail {
  float: right;
  margin-right: 20px;
}

.arrow-right {
  background-color: transparent;
  border: none;
  outline: none;
}

.arrow-right:hover {
  cursor: pointer;
}

.wrap {
  display: flex;
  flex-wrap: wrap;
}
