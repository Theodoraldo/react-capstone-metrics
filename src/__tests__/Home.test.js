import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Home from '../components/Home';

describe('Renders correctly', () => {
  test('renders the home component snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('object assignment', () => {
    render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>,
    );

    const data = { one: 1 };
    data.two = 2;
    data.three = 3;
    expect(data).toEqual({ one: 1, two: 2, three: 3 });
  });

  test('Continent data rendering', () => {
    render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>,
    );

    const continentList = [
      'Africa',
      'America',
      'Antarctic',
      'Asia',
      'Europe',
      'Oceania',
    ];

    const num = continentList.length;
    expect(num).toEqual(6);
    expect(continentList).toContain('Africa');
  });
});
