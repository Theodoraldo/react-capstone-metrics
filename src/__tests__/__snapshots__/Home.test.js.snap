// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly renders the home component snapshot 1`] = `
<DocumentFragment>
  <div
    class="home-header-container"
  >
    <p
      class="home-header"
    >
      The Worldâ€™s Continent
    </p>
  </div>
  <div
    class="search-head"
  >
    <input
      class="search"
      placeholder="Search country here ...."
      type="text"
    />
  </div>
  <div
    class="wrap"
  >
    <div
      class="continent"
    >
      <div
        class="cont-head"
      >
        <div>
          Africa
        </div>
        <button
          class="arrow-right"
          type="button"
        >
          <a
            href="/continent/Africa"
          >
            <svg
              color="#ececec"
              fill="currentColor"
              height="1.5rem"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(236, 236, 236);"
              viewBox="0 0 16 16"
              width="1.5rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </button>
      </div>
      <div
        class="cont-body"
      >
        <p
          class="cont-name"
        >
          Africa
        </p>
        <img
          alt="continent logo"
          src="africa.svg"
          width="200"
        />
      </div>
    </div>
    <div
      class="continent"
    >
      <div
        class="cont-head"
      >
        <div>
          America
        </div>
        <button
          class="arrow-right"
          type="button"
        >
          <a
            href="/continent/America"
          >
            <svg
              color="#ececec"
              fill="currentColor"
              height="1.5rem"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(236, 236, 236);"
              viewBox="0 0 16 16"
              width="1.5rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </button>
      </div>
      <div
        class="cont-body"
      >
        <p
          class="cont-name"
        >
          America
        </p>
        <img
          alt="continent logo"
          src="america.svg"
          width="200"
        />
      </div>
    </div>
    <div
      class="continent"
    >
      <div
        class="cont-head"
      >
        <div>
          Antarctic
        </div>
        <button
          class="arrow-right"
          type="button"
        >
          <a
            href="/continent/Antarctic"
          >
            <svg
              color="#ececec"
              fill="currentColor"
              height="1.5rem"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(236, 236, 236);"
              viewBox="0 0 16 16"
              width="1.5rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </button>
      </div>
      <div
        class="cont-body"
      >
        <p
          class="cont-name"
        >
          Antarctic
        </p>
        <img
          alt="continent logo"
          src="antarctica.svg"
          width="200"
        />
      </div>
    </div>
    <div
      class="continent"
    >
      <div
        class="cont-head"
      >
        <div>
          Asia
        </div>
        <button
          class="arrow-right"
          type="button"
        >
          <a
            href="/continent/Asia"
          >
            <svg
              color="#ececec"
              fill="currentColor"
              height="1.5rem"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(236, 236, 236);"
              viewBox="0 0 16 16"
              width="1.5rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </button>
      </div>
      <div
        class="cont-body"
      >
        <p
          class="cont-name"
        >
          Asia
        </p>
        <img
          alt="continent logo"
          src="asia.svg"
          width="200"
        />
      </div>
    </div>
    <div
      class="continent"
    >
      <div
        class="cont-head"
      >
        <div>
          Europe
        </div>
        <button
          class="arrow-right"
          type="button"
        >
          <a
            href="/continent/Europe"
          >
            <svg
              color="#ececec"
              fill="currentColor"
              height="1.5rem"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(236, 236, 236);"
              viewBox="0 0 16 16"
              width="1.5rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </button>
      </div>
      <div
        class="cont-body"
      >
        <p
          class="cont-name"
        >
          Europe
        </p>
        <img
          alt="continent logo"
          src="europe.svg"
          width="200"
        />
      </div>
    </div>
    <div
      class="continent"
    >
      <div
        class="cont-head"
      >
        <div>
          Oceania
        </div>
        <button
          class="arrow-right"
          type="button"
        >
          <a
            href="/continent/Oceania"
          >
            <svg
              color="#ececec"
              fill="currentColor"
              height="1.5rem"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(236, 236, 236);"
              viewBox="0 0 16 16"
              width="1.5rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </button>
      </div>
      <div
        class="cont-body"
      >
        <p
          class="cont-name"
        >
          Oceania
        </p>
        <img
          alt="continent logo"
          src="oceania.svg"
          width="200"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
